lib/controller.js

```diff
@@ -170,6 +177,134 @@ Tessel.get = function(opts) {
+
+    if (opts.timeout && typeof opts.timeout === 'number') {
+      // Stop the search after that duration
+      setTimeout(stopSeeker, opts.timeout * 1000);
+    } else {
+      // default to 3 seconds searching
+      setTimeout(stopSeeker, 3000);
+    }
+    logs.info('Searching accessable Tessels ...');
+
+    var seeker = new discover.TesselSeeker().start();
+    var tessels = [];
+
+    // When we find Tessels
+    seeker.on('tessel', function(tessel) {
+      controller.closeTesselConnections(tessel)
+        .then(function() {
+          var known = false;
+          if (tessels.length >= 1) {
+            for (var i in tessels) {
+              if (tessel.connections[0].ip === tessels[i].connections[0].ip && tessel.connections[0].connectionType === 'LAN') {
+                // Your network topology causes a single tessel is found twice
+                debug('known = true');
+                known = true;
+              }
+            }
+
+          }
+          if (known) {
+            debug('Due to your Network-Topology ' + tessel.name + ' is found twice! (' + tessel.connections[0].ip + ')');
+          } else {
+            // Add it to our array
+            tessels.push(tessel);
+          }
+        });
+    });
+
+    seeker.on('error', function(e) {
+      reject(e);
+    });
+
+    function stopSeeker() {
+      try {
+        // If there were no Tessels found
+        if (!tessels || tessels.length === 0) {
+          // Report the sadness
+          debug('We are searching the following networks:\n', networkInterfaces);
+          debug('\n(Important: Check your Netmask - we do not follow gateways for discovering!)');
+          logs.warn('No Tessels found (DEBUG=controller t2 root # might help!)');
+          resolve();
+        }
+        seeker.stop();
+        seeker = null; // preventing memory leaks
+        return resolve(tessels);
+      } catch (e) {
+        reject(e);
+      }
+    }
+  });
+};
+/*
+  The T2 root command is used to login into the firmware and gaining superuser access.
+  The security is provided by RSA - your Tessel get the keys while provisioning via USB.
+  
+  If you have only one Tessel, the T2 root command will login directly else there is 
+  a ncurses like menu you can select the Tessel you like to gain root access...
+
+  The structure of code and maybe unusual parts are paid due being testable.
+  The lightweight terminal-menu isn't written testable what needs a little bit hacking.
+
+  Finally some parts are causing from my personnal learning curve about writing unit tests using sinon!
+*/
+controller.root = function(opts) {
+  // ~ conversion to home because spawn isn't able to handle this right
+  if (opts.path && opts.path.substring(0, 1) === '~') {
+    var home = process.env.HOME;
+    opts.path = opts.path.replace('~', home);
+  }
+  var rtm;
+  if (!opts.menu) {
+    // TODO: Tessel cooperate identity conform menu required (color, ascii logo)
+    rtm = Menu({
```

Comment:
@johnnyman727 based on this ideas I am preparing an additional pull request, but I am not done at the moment.

What will be in:
- [x] makeCommand('select')
- [x] Tessel.seekTessels() -> returns list of found Tessels **asynchronously!**
- [x] controller.ssh.checkAuthorized() **asynchronously!**
- [x] testSSHConnection() **asynchronously!**
- [x] getAuthorizedTessels() with async.forEachOf(tessels, testSSHConnection)  **asynchronously!**
- [x] getTerminalsPID() **asynchronously!**
- [x] checkPIDexists(pid) process.platform based **asynchronously!**
- [x] checkAllPIDfiles() process.platform based **asynchronously!**
- [ ] createSelectedTesselPIDfile(pid) **asynchronously!**
- [x] checkAllPIDfiles() fs.readdirAsync & fs.readFileAsync **asynchronously!**
- [ ] set process.env.Tessel
- [ ] get global id_rsa
- [ ] let it work with menu pull request
```
var authorizedTessels = [];
    if (opts.name) {
      // if name selected test if authorized
      testSSHConnection()
      .getTerminalsPID()
      .checkAllPIDfiles()
      .createSelectedTesselPIDfile();
      //.catch()
    } else {
      getTerminalsPID()
      .checkAllPIDfiles()
      .getAuthorizedTessels(function(tessels) {
        if(tessels.length === 0) {
          // no authorized Tessels
          reject('No authorized Tessels found!');
        } else if (tessels.length === 1) {
          // select direct
          createSelectedTesselPIDfile();
        } else {
          // menu
          controller.menu.prepare(tessels)
          .createSelectedTesselPIDfile();
        }
      })
      .controller.menu.prepare()
      .createSelectedTesselPIDfile();
    }
```